charts/elasticsearch/templates/data/es-data-statefulset.yaml

```diff
@@ -2,7 +2,11 @@
 ## Elasticsearch Data StatefulSet
 #################################
 kind: StatefulSet
+{{- if gt (.Capabilities.KubeVersion.Minor | atoi) 15 }}
+apiVersion: apps/v1
+{{- else }}
 apiVersion: apps/v1beta2
+{{- end }}
```

Comment:
this is how to split api b/w kube versions. I tried to avoid this when possible, and I'm using 'greater than' so it doesn't require more work moving forward into future versions. We only need to check the minor version because the minimum version is specified in Chart.yaml to >= 1.12 (so we don't need to worry about major version 0)